src/main/resources/assets/opencomputers/loot/openos/lib/serialization.lua

```diff
@@ -1,134 +1,93 @@
-        else
-          if tk == "string" and not kw[k] and string.match(k, id) then
-            r = r .. k
-          else
-            r = r .. "[" .. s(k, l + 1) .. "]"
-          end
-          r = r .. "=" .. s(v, l + 1)
-        end
-      end
-      ts[v] = nil -- allow writing same table more than once
-      return (r or "{") .. "}"
-    else
-      if pretty then
-        return tostring(v)
-      else
-        error("unsupported type: " .. t)
-      end
-    end
-  end
-  local result = s(value, 1)
-  local limit = type(pretty) == "number" and pretty or 10
-  if pretty then
-    local truncate = 0
-    while limit > 0 and truncate do
-      truncate = string.find(result, "\n", truncate + 1, true)
-      limit = limit - 1
-    end
-    if truncate then
-      return result:sub(1, truncate) .. "..."
-    end
-  end
-  return result
+------------------------------------------------- Private methods -----------------------------------------------------------------
+
+local function doSerialize(array, text, prettyLook, indentationSymbol, oldIndentationSymbol, equalsSymbol)
+	text = {"{"}
+	table.insert(text, (prettyLook and "\n" or nil))
+	
+	for key, value in pairs(array) do
+		local keyType, valueType, stringValue = type(key), type(value), tostring(value)
+
+		if keyType == "number" or keyType == "string" then
+			table.insert(text, (prettyLook and indentationSymbol or nil))
+			table.insert(text, "[")
+			table.insert(text, (keyType == "string" and table.concat({"\"", key, "\""}) or key))
+			table.insert(text, "]")
+			table.insert(text, equalsSymbol)
+			
+			if valueType == "number" or valueType == "boolean" or valueType == "nil" then
+				table.insert(text, stringValue)
+			elseif valueType == "string" or valueType == "function" then
+				table.insert(text, "\"")
+				table.insert(text, stringValue)
+				table.insert(text, "\"")
+			elseif valueType == "table" then
+				table.insert(text, table.concat(doSerialize(value, text, prettyLook, table.concat({indentationSymbol, indentationSymbol}), table.concat({oldIndentationSymbol, indentationSymbol}), equalsSymbol)))
+			else
+				error("Unsupported table value type: " .. valueType)
+			end
+			
+			table.insert(text, ",")
+			table.insert(text, (prettyLook and "\n" or nil))
+		else
+			error("Unsupported table key type: " .. keyType)
+		end
+	end
+
+	table.remove(text, (prettyLook and #text - 1 or #text))
+	table.insert(text, (prettyLook and oldIndentationSymbol or nil))
+	table.insert(text, "}")
+	return text
+end
+
+------------------------------------------------- Public methods -----------------------------------------------------------------
+
+function serialization.serialize(array, prettyLook, indentationWidth, indentUsingTabs)
+	checkArg(1, array, "table")
+	indentationWidth = indentationWidth or 2
+	local indentationSymbol = indentUsingTabs and "	" or " "
+	indentationSymbol, indentationSymbolHalf = string.rep(indentationSymbol, indentationWidth)
```

Comment:
I checking this code via browser, not with an editor, nor have I tested. But -- is that " "(space) or " "(space) ? Is "\t" what you meant? Also, what is indentationSymbolHalf used for here? string.rep only returns 1 value. Also, if the user sends using tabs true and width 4, is that 4 tabs? Is that what we want?